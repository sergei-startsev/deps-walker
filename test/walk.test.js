const path = require('path');
const depsWalker = require('../lib');

const convertToRelative = filePath =>
  filePath
    .toLowerCase()
    .replace(path.resolve(__dirname, '..').toLowerCase(), '.')
    .replace(/\\/g, '/');

describe('walker', function () {
  it('should traverse an entry with default configuration', async function () {
    const walk = depsWalker();

    const graph = await walk(
      path.resolve('./examples/base/entry.js'),
      () => {}
    );

    const expected = graph.map(({ filePath, dependencies }) => {
      filePath = convertToRelative(filePath);
      dependencies = dependencies.map(convertToRelative);
      return { filePath, dependencies };
    });
    expect(expected).toMatchSnapshot();
  });

  it('should traverse several entries with default configuration', async function () {
    const walk = depsWalker();

    const graph = await walk(
      [
        path.resolve('./examples/base/a.js'),
        path.resolve('./examples/base/entry.js')
      ],
      () => {}
    );
    const expected = graph.map(({ filePath, dependencies }) => {
      filePath = convertToRelative(filePath);
      dependencies = dependencies.map(convertToRelative);
      return { filePath, dependencies };
    });
    expect(expected).toMatchSnapshot();
  });
});
