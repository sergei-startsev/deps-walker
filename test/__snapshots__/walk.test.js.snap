// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walker should traverse an entry with default configuration 1`] = `
[
  {
    "dependencies": [
      "./test/examples/base/a.js",
      "./test/examples/base/b.js",
    ],
    "filePath": "./test/examples/base/entry.js",
  },
  {
    "dependencies": [
      "./test/examples/base/b.js",
      "./test/examples/base/c.js",
      "./test/examples/base/d.js",
    ],
    "filePath": "./test/examples/base/a.js",
  },
  {
    "dependencies": [
      "./test/examples/base/d.js",
    ],
    "filePath": "./test/examples/base/c.js",
  },
  {
    "dependencies": [
      "./test/examples/base/b.js",
    ],
    "filePath": "./test/examples/base/d.js",
  },
  {
    "dependencies": [],
    "filePath": "./test/examples/base/b.js",
  },
]
`;

exports[`walker should traverse several entries with default configuration 1`] = `
[
  {
    "dependencies": [
      "./test/examples/base/b.js",
      "./test/examples/base/c.js",
      "./test/examples/base/d.js",
    ],
    "filePath": "./test/examples/base/a.js",
  },
  {
    "dependencies": [
      "./test/examples/base/d.js",
    ],
    "filePath": "./test/examples/base/c.js",
  },
  {
    "dependencies": [
      "./test/examples/base/b.js",
    ],
    "filePath": "./test/examples/base/d.js",
  },
  {
    "dependencies": [
      "./test/examples/base/a.js",
      "./test/examples/base/b.js",
    ],
    "filePath": "./test/examples/base/entry.js",
  },
  {
    "dependencies": [],
    "filePath": "./test/examples/base/b.js",
  },
]
`;
